### PR TITLE
vue 2.x to vue 3.x

### PR DESCRIPTION
**Pull request by marvelbark2**
Updating vue APi with the new APIs

### CODE REVIEW

File: src/FullCalendar.ts [hunk 1]
```diff
@@ -12,16 +12,16 @@ interface FullCalendarInternal {
 }
 
 
-const FullCalendar = Vue.extend({
+const FullCalendar = Vue.extend{
```
Comment (marvelbark2) [inline src/FullCalendar.ts:15]: Vue.extend suppose to be deleted because its replaced by defineComponent

**Comment by acerix** [comment]
Thanks, this fixes https://github.com/fullcalendar/fullcalendar-vue/issues/131 ?

**Comment by jippeholwerda** [comment]
I would expect the version of Vue to be updated in `package.json` as well?

**Comment by aliasaria** [comment]
This doesn't build for me:
`npm run build`

results in:

```bash
> tsc -p tsconfig.json && rollup -c && npm run minify

src/FullCalendar.ts:15:32 - error TS1005: ',' expected.

15 const FullCalendar = Vue.extend{
                                  ~

src/FullCalendar.ts:18:21 - error TS1005: ',' expected.

18     options: Object as PropType<CalendarOptions>
                       ~~

src/FullCalendar.ts:18:24 - error TS1005: ',' expected.
...
```

**Comment by marvelbark2** [comment]
> ```shell
> Vue.extend
> ```

Hey , You should delete 
`Vue.extend` on `const FullCalendar = ...`

**Comment by aliasaria** [comment]
@marvelbark2 do you mean you can get `npm run build` to build this project? Removing this doesn't fix the build. I am just building a plain copy of your code:

https://github.com/marvelbark2/fullcalendar-vue/blob/patch-2/src/FullCalendar.ts

**Comment by marvelbark2** [comment]
@aliasaria  I fixed the issue !

**Comment by arshaw** [comment]
note to self: releasing this will break compatibility with Vue 2. release this as a version like `5.1.2-vue3`, then have a dist-tag called `vue3` that makes npm-installing more convenient.

any input on a better release strategy is welcome

**Comment by SKFrozenCloud** [comment]
That sounds good

**Comment by potraike** [comment]
Hi, do you have any plans when this release will be available?

**Comment by guifischer** [comment]
Any chance to get a vue 3 "beta" branch with this changes?

**Comment by arshaw** [comment]
I've merged this into the `vue3` branch and released it as the `@fullcalendar/vue3` package!

**Comment by arshaw** [comment]
(it required quite a bit more work), but thanks so much for helping me get started!

**Pull request closed without merging**